docs/samples/Microsoft.ML.Samples/Dynamic/BinaryClassification/AveragedPerceptron.cs

```diff
@@ -0,0 +1,61 @@
+﻿using Microsoft.ML;
+
+namespace Microsoft.ML.Samples.Dynamic.BinaryClassification
+{
+    public static class AveragedPerceptron
+    {
+        public static void Example()
+        {
+            // In this examples we will use the adult income dataset. The goal is to predict
+            // if a person's income is above $50K or not, based on different pieces of information about that person.
+            // For more details about this dataset, please see https://archive.ics.uci.edu/ml/datasets/adult
+
+            // Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
+            // as a catalog of available operations and as the source of randomness.
+            // Setting the seed to a fixed number in this examples to make outputs deterministic.
+            var mlContext = new MLContext(seed: 0);
+
+            // Download the dataset and load it as IDataView
+            var data = SamplesUtils.DatasetUtils.LoadAdultDataset(mlContext);
+
+            // Leave out 10% of data for testing
+            var (trainData, testData) = mlContext.BinaryClassification.TrainTestSplit(data, testFraction: 0.1);
+
+            // Create data processing pipeline
+            var pipeline =
+                // Convert categorical features to one-hot vectors
+                mlContext.Transforms.Categorical.OneHotEncoding("workclass")
```

Comment:
The featurization here is a bit complex for discussing a learner. Can you please move this into the `LoadAdultDataset` method, with an optional parameter (default `false`) that applies this featurization?

Reply:
Can you start on the same line as `pipeline`?

Reply:
i want to put // comment for each section of the pipeline. . I can't add (// Convert categorical features to one-hot vectors) if I put in the same line. 

---
In reply to: [255265735](https://github.com/dotnet/machinelearning/pull/2483#discussion_r255265735) [](ancestors = 255265735)

Reply:
OneHotEncoding is the minimum we need for handling text features. we can't really use this dataset otherwise. I'm moving the pipeline to utils, but the trainer can't get an acceptable AUC by making this transform optional. agree?

---
In reply to: [255264730](https://github.com/dotnet/machinelearning/pull/2483#discussion_r255264730) [](ancestors = 255264730)